.github/workflows/pull_request.yaml

```diff
@@ -9,28 +9,22 @@ jobs:
     runs-on: ubuntu-22.04
     steps:
     - name: Cancel Previous Runs
-      uses: styfle/cancel-workflow-action@0.4.0
+      uses: styfle/cancel-workflow-action@0.11.0
       with:
         access_token: ${{ github.token }}
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
     - name: Set up Go 1.14
-      uses: actions/setup-go@v1
+      uses: actions/setup-go@v4
       with:
-        go-version: 1.14
+        go-version-file: "go.mod"
       id: go
     - name: Install Protoc
-      uses: solo-io/setup-protoc@master
+      uses: arduino/setup-protoc@master
       with:
         version: '3.6.1'
         repo-token: ${{ secrets.GITHUB_TOKEN }}
-    - uses: actions/checkout@v2.4.1
-      with:
-        fetch-depth: 0
-    - uses: actions/cache@v1
-      with:
-        path: ~/go/pkg/mod
-        key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
-        restore-keys: |
-          ${{ runner.os }}-go-
```

Comment:
why are we getting rid of the actions/cache?

Reply:
Actually I think we were not using the cache correctly, but if we can it would be nice.
Since this is the code that generates code, I do not think that caching would help actually.|
Reading what `actions/setup-go@v4` does it looks like the `go mod download` is cached.

https://github.com/actions/setup-go

Please include this in the description of the PR.

Reply:
Super helpful docs thank you!